services/dataset.js

```diff
@@ -134,9 +168,49 @@ export const deleteDataset = (id, token) => {
     });
 };
 
+/**
+ * Updates or creates a metadata object
+ * This methods requires authentication.
+ *
+ * @param {string} type - one of the following: 'PATCH', 'POST'
+ * @param {Object} data - metadata object
+ * @param {*} id - dataset ID
+ * @param {string} token - user's token.
+ * @returns {Object} New or updated metadata object.
+ */
+export const saveMetadata = ({ type, data, id = '', token }) => {
```

Comment:
I would split this into two functions (following CRUD).. Functions doing different stuff only leads to bugs and harder to mantain.

Reply:
@andresgnlez Similarly to what I posted in a comment above. Let's create a separate task if necessary to improve the dataset service. Otherwise we risk making too many not-so-much-related changes as part of this PR.